### ISSUE TITLE
False positive for `operator_usage_whitespace` and generic parameters

### ISSUE THREAD

**Issue by jessesquires** [realm/SwiftLint#3668]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

Related: #2728

### Describe the bug

Rule `operator_usage_whitespace` produces a false positive in the following scenario:

```swift
protocol TProtocol { }

protocol UProtocol { }

struct Model {
    init<T: TProtocol,                    // the `<` triggers operator_usage_whitespace
         U: UProtocol>(one: T, two: U) {  // the `>` triggers operator_usage_whitespace

    }
}
```

SwiftLint complains about the `<` and `>`.

If the generic parameters are placed on the same line, the bug does not happen:

```swift
struct Model {
    init<T: TProtocol, U: UProtocol>(one: T, two: U) {

    }
}
```

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? **0.43.1**
* Installation method used (Homebrew, CocoaPods, building from source, etc)? **Homebrew**
* Paste your configuration file:

```yml
opt_in_rules:
    - operator_usage_whitespace
```

### Which Xcode version are you using (check `xcodebuild -version`)?
- Version 12.5 (12E262)
- Version 13.0 beta (13A5155e)

**Comment by mikchmie** [comment]
Same happens in property definitions when type names are long and you try to adhere to line length limit:
```
private let someVeryLongProperty = MyGenericClassWithAVeryLongName<VeryLongGenericParamA,   // the `<` triggers operator_usage_whitespace
                                                                   VeryLongGenericParamB>() // the `>` triggers operator_usage_whitespace
```
